### ISSUE TITLE
Restore custom tooltips

### ISSUE THREAD

**Issue by gr455** [CircuitVerse/CircuitVerse#2887]
**Describe the bug**
Due to some error (presumably in jquery-ui) custom tooltips have stopped working and result in unhandled exceptions. They have been temporarily removed in [this](https://github.com/CircuitVerse/CircuitVerse/pull/2886/commits/ba929958c844e3c4d319ac7dcf2480fe56332945) commit. We need to fix the problem and restore the custom tooltips.

**Expected behavior**
Custom tooltips must work on hovering over components (say a circuit element in the module list)

**Device Information [optional]:**
 - Browser: Chrome
 - Version: 96.0.4664.110

**Additional context**
Have a look at how the custom tooltips were initially implemented https://github.com/CircuitVerse/CircuitVerse/pull/2886/commits/ba929958c844e3c4d319ac7dcf2480fe56332945

**Comment by gr455** [comment]
Called tooltip on a specific selector instead of `$(document)`. Seems to get rid of the errors, but no custom styling is applied to the tooltips

**Comment by vedant-jain03** [comment]
Can we use stimulus for this, as now stimulus is powered in our whole application

**Comment by gr455** [comment]
> @gr455 Can we use stimulus for this, as now stimulus is powered in our whole application

I am not too familiar with stimulus, but if it can get the job done, it would be great!

**Comment by Dnouv** [comment]
Hello @gr455 @vedant-jain03 
Is this still available to contribute? I would like to work on this issue.
I am looking forward to hearing back from you. Thank you!

**Comment by vedant-jain03** [comment]
Sorry for responding late, @Dnouv You can try this issue. I will try to solve it using stimulus and get back in this thread .

**Comment by Dnouv** [comment]
Thank you for your prompt reply @vedant-jain03 . Could you please let me know what was the expected behavior because, on the simulator page, the custom tooltip is appearing on hover, here is an attached clip of the tooltip,
![chrome-capture-2022-1-23](https://user-images.githubusercontent.com/61188295/155313083-9716bf3f-360e-4add-9ffd-82a0872805c5.gif)

**Comment by vedant-jain03** [comment]
> Thank you for your prompt reply @vedant-jain03 . Could you please let me know what was the expected behavior because, on the simulator page, the custom tooltip is appearing on hover, here is an attached clip of the tooltip, ![chrome-capture-2022-1-23](https://user-images.githubusercontent.com/61188295/155313083-9716bf3f-360e-4add-9ffd-82a0872805c5.gif)

@gr455

**Comment by Dnouv** [comment]
@vedant-jain03 any updates on this issue?

**Comment by Dnouv** [comment]
Hello @gr455 
I understand your busy schedule but could you please verify this issue once and let me know the expected behavior.
Thank you!

**Comment by gr455** [comment]
> Thank you for your prompt reply @vedant-jain03 . Could you please let me know what was the expected behavior because, on the simulator page, the custom tooltip is appearing on hover, here is an attached clip of the tooltip, ![chrome-capture-2022-1-23](https://user-images.githubusercontent.com/61188295/155313083-9716bf3f-360e-4add-9ffd-82a0872805c5.gif)

Yep. this is happening on the master branch. The latest state of the master branch is not yet deployed, so this won't be seen on circuitverse.org.

I removed the code for the custom tooltips because it was throwing unhandled errors in the commit referenced in the issue.

**Comment by Dnouv** [comment]
No worries at all, yeah I just re-checked and the tooltip (as appearing on the circuitverse.org) is not appearing on the development server, so I guess we need to show a custom tooltip something like in the above video. I'll try to resolve this issue.
Thank you!

**Comment by kushalShukla-web** [comment]
can i work on this issue

**Comment by Dnouv** [comment]
Hello @kushalShukla-web sorry but I'm working on this issue.
Thank you!

**Comment by tachyons** [comment]
Sure

**Comment by Pulkit0729** [comment]
I see the issue already has a pr, but codeclimate requested pr changes, which are still pending, i will be adding a new Pr with final changes.

**Comment by Pulkit0729** [comment]
cc @vedant-jain03 , @tachyons , @gr455

**Comment by mounika63020-blip** [comment]
Hello everyone,

My name is Mounika Kappala. I am a student from India and I am very interested in contributing to open-source projects. I have knowledge of HTML, CSS, and JavaScript, and I am currently learning more about web development and digital electronics.

I would like to start contributing to this project and help improve CircuitVerse. I am especially interested in working on beginner-friendly issues such as UI improvements or bug fixes. Please guide me on how I can get started with this issue.

Thank you!
